src/app/core/models/label.model.ts

```diff
@@ -1,4 +1,28 @@
 export class Label {
+
+  // Preset Labels.
+  private static readonly REQUIRED_LABELS = {
+    severity: {
+      Low: new Label('severity', 'Low', 'ffb3b3'),
+      Medium: new Label('severity', 'Medium', 'ff6666'),
+      High: new Label('severity', 'High', 'b30000')
+    },
```

Comment:
Not sure if it is a good practice to define attributes that require the constructor method of the encapsulating class. Prof @damithc, what do you think about it?

Maybe it would be better to extract it out to a service class. It would be easier to extract those string values to a configuration file in the future too.

Reply:
> Not sure if it is a good practice to define attributes that require the constructor method of the encapsulating class. Prof @damithc, what do you think about it?

not necessarily a problem. For example, the Singleton pattern uses the constructor of the class within the class itself.

>Maybe it would be better to extract it out to a service class. It would be easier to extract those string values to a configuration file in the future too.

Yes, in this case, these attributes are better put in a service class rather than the Label class. Reason: these attributes indicate _how_ the App _uses_ the Label class, rather than being an integral part of a Label.

Reply:
> Yes, in this case, these attributes are better put in a service class rather than the Label class. Reason: these attributes indicate _how_ the App _uses_ the Label class, rather than being an integral part of a Label.

Sure. I will shift it out to `label.service.ts`.